packages/gatsby-theme-sso-ui-kit/src/utils/analytics.utils.ts

```diff
@@ -0,0 +1,87 @@
+// utility to track GA events client-side in a robust ways
+// lives in the SSO theme because typescript support is helpful in this case and the event tracking use cases are
+// (until the packages are merged anyways) all learning and authentication related
+// Recommended events reference: https://developers.google.com/tag-platform/gtagjs/reference/events
+
+export const gtagEvent = (
+  eventName: EventNames | (string & {}),
```

Comment:
EventNames type doesn't include our own event names such as `submit_quiz`. I feel we should either add them to the `EventNames` type or create a `SelfLearningEventNames`.

Reply:
good idea, done !